src/components/manager.ts

```diff
@@ -87,6 +85,23 @@ export class Manager {
         return path.normalize(out).split(path.sep).join('/')
     }
 
+    /**
+     * The indicator of the root file.
+     */
+    get rootIndictor() {
+        const configuration = vscode.workspace.getConfiguration('latex-workshop')
+        const indicator = configuration.get('latex.rootFile.indicator')
+        switch (indicator) {
+            case '\\documentclass[]{}':
+                return /\\documentclass(?:\s*\[.*\])?\s*\{.*\}/ms
+            case '\\begin{document}':
+                return /\\begin{document}/m
```

Comment:
It's possible to have spaces or newlines before the left brace `{`. This case was neglected in previous versions. Please add this.

Reply:
Sorry, I am not familiar with regex. These are directly copied from the source code. I don't know how to modify them. Could you please provide  me the required regex?

Reply:
You may try /\\begin\s*{document}/m